ip_test.go

```diff
@@ -0,0 +1,214 @@
+	server, client, mux := setupTestClient()
+	defer server.Close()
+	uri := path.Join(apiNetworkBase, dummyUuid, "events")
+	mux.HandleFunc(uri, func(writer http.ResponseWriter, request *http.Request) {
+		assert.Equal(t, http.MethodGet, request.Method)
+		fmt.Fprintf(writer, prepareIpEventListHTTPGet())
+	})
+	res, err := client.GetIpEventList(dummyUuid)
+	if err != nil {
+		t.Errorf("GetIpEventList returned an error %v", err)
+	}
+	assert.Equal(t, 1, len(res))
+	assert.Equal(t, fmt.Sprintf("[%v]", getMockIpEvent()), fmt.Sprintf("%v", res))
+}
+
+func TestClient_GetIpVersion(t *testing.T) {
+	server, client, mux := setupTestClient()
+	defer server.Close()
+	uri := path.Join(apiIpBase, dummyUuid)
+	mux.HandleFunc(uri, func(writer http.ResponseWriter, request *http.Request) {
+		assert.Equal(t, http.MethodGet, request.Method)
+		fmt.Fprintf(writer, prepareIpHTTPGet())
+	})
+	res := client.GetIpVersion(dummyUuid)
+	if res == 0 {
+		t.Error("GetIpVersion has an error")
+	}
+	assert.Equal(t, 1, res)
+
+}
+
+func getMockIP() Ip {
+	mock := Ip{Properties:IpProperties{
+		Name:            "test",
+		LocationCountry: "Germany",
+		LocationUuid:    dummyUuid,
+		ObjectUuid:      dummyUuid,
+		ReverseDns:      "8.8.8.8",
+		Family:          1,
+		Status:          "active",
+		CreateTime:      dummyTime,
+		Failover:        false,
+		ChangeTime:      dummyTime,
+		LocationIata:    "",
+		LocationName:    "Cologne",
+		Prefix:          "",
+		Ip:              "192.168.0.1",
+		DeleteBlock:     "",
+		UsagesInMinutes: 10,
+		CurrentPrice:    0.9,
+		Labels:          []string{"label"},
+		Relations:       IpRelations{
+			Loadbalancers: []IpLoadbalancer{
+				{
+					CreateTime:       dummyTime,
+					LoadbalancerName: "test",
+					LoadbalancerUuid: dummyUuid,
+				},
+			},
+		},
+	}}
+	return mock
+}
+
+func prepareIpListHTTPGet() string {
+	ip := getMockIP()
+	res, _ := json.Marshal(ip.Properties)
+	return fmt.Sprintf(`{"ips": {"%s": %s}}`, dummyUuid, string(res))
+}
+
+func prepareIpHTTPGet() string {
+	ip := getMockIP()
+	res, _ := json.Marshal(ip)
+	return string(res)
+}
+
+func getMockIpCreateResponse() IpCreateResponse {
+	mock := IpCreateResponse{
+		RequestUuid: dummyRequestUUID,
+		ObjectUuid:  dummyUuid,
+		Prefix:      "ip",
+		Ip:          "192.168.0.1",
+	}
+	return mock
+}
+
+func prepareIpCreateResponse() string {
+	res, _ := json.Marshal(getMockIpCreateResponse())
+	return string(res)
+}
+
+func getMockIpEvent() IpEvent {
+	mock := IpEvent{Properties:IpEventProperties{
+		ObjectType:    "type",
+		RequestUuid:   dummyRequestUUID,
+		ObjectUuid:    dummyUuid,
+		Activity:      "activity",
+		RequestType:   "tcp",
+		RequestStatus: "done",
+		Change:        "change note",
+		Timestamp:     dummyTime,
+		UserUuid:      dummyUuid,
+	}}
+	return mock
+}
+
+func prepareIpEventListHTTPGet() string {
+	event := getMockIpEvent()
+	res, _ := json.Marshal(event.Properties)
+	return fmt.Sprintf(`{"events": [%s]}`, string(res))
+}
```

Comment:
a new line is missing.

Reply:
What do you mean by "a new line is missing", please?